src/de.rs

```diff
@@ -448,10 +451,17 @@ impl<'de> de::Deserializer<'de> for Deserializer {
     /// Simply calls `visit_newtype_struct`.
     fn deserialize_newtype_struct<V: de::Visitor<'de>>(
         self,
-        _name: &'static str,
+        name: &'static str,
         visitor: V,
     ) -> Result<V::Value> {
-        visitor.visit_newtype_struct(self)
+        if name == PRESERVED_VALUE_MAGIC {
+            let de = Deserializer {
```

Comment:
Hm this looks like no-op? It's turning a value into ABI and then back.

Reply:
Not quite, it's turning a `JsValue { index }` (i.e. a JS object living at index `index` in the wasm-bindgen heap) into a JS object representing a number with value `index`.

Actually, maybe this is an argument in favor of dropping the wrapper struct, because this is forced by the autoderived `Visitor`, which generates a deserialization error if we return `visit.visit_u32(..)`